pkg/objectupdate/rte/rte_test.go

```diff
@@ -179,14 +179,8 @@ func TestUpdateDaemonSetArgs(t *testing.T) {
 }
 
 func expectCommandLine(t *testing.T, ds, origDs *appsv1.DaemonSet, testName string, expectedArgs []string) {
-	uncontrolledArgs := []string{
-		//	"--notify-file=/run/rte/notify",
-		"--podresources-socket=unix:///host-var/lib/kubelet/pod-resources/kubelet.sock",
-		"--sysfs=/host-sys",
-		"--topology-manager-policy=restricted",
-		"--topology-manager-scope=pod",
-	}
-	if len(expectedArgs) != len(ds.Spec.Template.Spec.Containers[0].Args)-len(uncontrolledArgs) {
+	expectedArgs=append(expectedArgs,commonArgs...)
```

Comment:
I need to check if this idiom can leak the appended args in the original slice. I think in the common case it will not, but in some unlikely but possible corner cases it can.
the safest (and clunkiest) way is to clone `expectedArgs`:
```
expectedArgs = append([]string{}, expecredArgs...)
expectedArgs = append(expectedArgs, commonArgs...)
```

(can be done without variable shadowing in the first line).

Likely not worth resubmitting for this, though.

Reply:
interesting, I'll validate. thanks for rasing that